src/assets/RestApi/rest-api.md

```diff
@@ -0,0 +1,400 @@
+---
+title: Understanding REST APIs - A Comprehensive Guide with Practical Example
+authorName: Prajwol Amatya
+authorAvatar: https://1.gravatar.com/avatar/de64e53c0e2cb393dd0d14ffdd53058ee9c607b35e366dd392425bd1b95a034c?size=256
+authorLink: https://github.com/prajwolamatya
+createdAt: April 16, 2024
+tags: rest, api, rest api, python, flask
+banner: https://blog.jankaritech.com/src/assets/RestApi/images/rest-api-banner.png
+---
+
+Within the field of web development, APIs (Application Programming Interface) are essential for facilitating communication across various software components. Because of its simplicity, scalability, and statelessness, REST (Representational State Transfer) APIs stand out among the other types of APIs. In this blog, we will dive into basics of building a REST API.
+
+## What is a REST API?
+REST is a framework that outlines a collection of guidelines for building web services. A REST API offers a straightforward and adaptable method to interact with web services without the need for any processing. REST is often favored for its lower bandwidth usage, simplicity, and flexibility, making it better suited for internet applications. REST is utilized to retrieve or transmit data from a web service with all interactions through a REST API occurring via HTTP requests.
```

Comment:
I don't think we can say REST is a framework but an architecture style. Ref:https://ics.uci.edu/~fielding/pubs/dissertation/rest_arch_style.htm